support-frontend/assets/components/csr/csrBanner.tsx

```diff
@@ -17,18 +17,23 @@ const banner = css`
 	z-index: 999;
 `;
 
-function CsrBanner() {
-	const [csrCustomerData, setCsrCustomerData] = useState<
-		CsrCustomerData | undefined
-	>(undefined);
+interface CsrBannerProps {
+	csrCustomerData: CsrCustomerData;
+	setCsrCustomerData: ActionCreatorWithPayload<CsrCustomerData>;
+}
+
+function CsrBanner({
+	csrCustomerData,
+	setCsrCustomerData,
+}: CsrBannerProps): JSX.Element | null {
```

Comment:
If we're providing data from the store as props it would be better to use `mapStateToProps` and `mapDispatchToProps`- this would also remove the need to inject `<CsrBanner />` as a prop into the `Page` component, and means that `setCsrCustomerData` will be wrapped with `dispatch` and will actually trigger a state update when it's called- passing in the action creator by itself means all you have is a function that returns an action object.

```suggestion
function mapStateToProps(state: SubscriptionsState) {
	return {
		csrCustomerData: state.page.checkoutForm.csrCustomerData,
	}
}

const mapDispatchToProps = {
	setCsrCustomerData,
}

const connector = connect(mapStateToProps, mapDispatchToProps);

type CsrBannerProps = ConnectedProps<typeof connector>;

// component code

export default connector(CsrBanner);
```